be/src/olap/rowset/segment_v2/segment.cpp

```diff
@@ -101,7 +110,8 @@ Status Segment::new_iterator(const Schema& schema, const StorageReadOptions& rea
             continue;
         }
         int32_t uid = read_options.tablet_schema->column(column_id).unique_id();
-        if (_column_readers.count(uid) < 1 || !_column_readers.at(uid)->has_zone_map()) {
+        if (_column_readers.count(uid) < 1 || !_column_readers.at(uid)->has_zone_map() ||
+            cond_in_compound_query(column_id)) {
```

Comment:
Is it necessary to add this check?

If a column is in remaininig compound predicates, zone map should take effect to prune segment as normal.

Reply:
It is unnecessary in the latest code implementation and has been removed